actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -379,7 +379,7 @@ def per_form_csrf_token(session, action_path, method)
 
       def xor_byte_strings(s1, s2)
         s2_bytes = s2.bytes
-        s1.bytes.map.with_index { |c1, i| c1 ^ s2_bytes[i] }.pack('c*')
+        s1.bytes.each_with_index { |c1, i| c1 ^ s2_bytes[i] }.pack('c*')
```

Comment:
`#each_with_index` returns the original collection, not the results of the block called on each element, so this change should cause tests to fail.

```ruby
>> [1,2,3].each_with_index { |e,i| e * i }.pack('c*')
=> "\x01\x02\x03"
>> [1,2,3].map.with_index { |e,i| e * i }.pack('c*')
=> "\x00\x02\x06"
```